lib/awful/widget/inputbox.lua

```diff
@@ -0,0 +1,530 @@
+                    local text_end = self._private.text:sub(self._private.highlight.cur_pos_end + 1)
+                    self._private.text = text_start .. text_end
+                    self._private.highlight = {}
+                    cursor_pos = #text_start + 1
+                else
+                    if cursor_pos > 1 then
+                        self._private.text = self._private.text:sub(1, cursor_pos - 2) ..
+                            self._private.text:sub(cursor_pos)
+                        cursor_pos = cursor_pos - 1
+                    end
+                end
+            elseif key == "Delete" then
+                -- If text is highlighted delete that, else just delete the character to the right
+                if self._private.highlight and self._private.highlight.cur_pos_start and
+                    self._private.highlight.cur_pos_end then
+                    local text_start = self._private.text:sub(1, self._private.highlight.cur_pos_start - 1)
+                    local text_end = self._private.text:sub(self._private.highlight.cur_pos_end + 1)
+                    self._private.text = text_start .. text_end
+                    self._private.highlight = {}
+                    cursor_pos = #text_start + 1
+                else
+                    if cursor_pos <= #self._private.text then
+                        self._private.text = self._private.text:sub(1, cursor_pos - 1) ..
+                            self._private.text:sub(cursor_pos + 1)
+                    end
+                end
+            elseif key == "Left" then
+                -- Move cursor ro the left
+                if cursor_pos > 1 then
+                    cursor_pos = cursor_pos - 1
+                end
+                self._private.highlight = {}
+            elseif key == "Right" then
+                -- Move cursor to the right
+                if cursor_pos <= #self._private.text then
+                    cursor_pos = cursor_pos + 1
+                end
+                self._private.highlight = {}
+            else
+                -- Print every alphanumeric key
+                -- It seems like gears.xmlescape doesn't support non alphanumeric characters
+                if key:wlen() == 1 then
+                    self._private.text = self._private.text:sub(1, cursor_pos - 1) ..
+                        key .. self._private.text:sub(cursor_pos)
+                    cursor_pos = cursor_pos + 1
+                end
+            end
+
+            -- Make sure the cursor cannot go out of bounds
+            if cursor_pos < 1 then
+                cursor_pos = 1
+            elseif cursor_pos > #self._private.text + 1 then
+                cursor_pos = #self._private.text + 1
+            end
+        end
+
+        -- Update cycle
+        self:update(self._private.text, cursor_pos)
+
+        if callbacks.pressed then
+            callbacks.pressed(key, mod_keys)
```

Comment:
Users are already used to the reverse order of `mods, key` from `awful.key`. That shouldn't change here.